### DISCUSSION TITLE
Doing work after the HTTP response has been sent

### DISCUSSION THREAD

**Discussion by lsmith77** [fastapi/fastapi#6578] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request

app = FastAPI()

@app.middleware("http")
async def write_log(request: Request, call_next):
    response = await call_next(request)
    # write log
    return response
```

### Description

I want to log data for each request, however since my application is latency sensitive, I would want to return as quickly as possible. Is there an equivalent to Symfony's "[terminate](https://symfony.com/doc/current/reference/events.html#kernel-terminate)" event (which I guess is the `request_finished` signal in Django)? The idea is to do the log writing after the HTTP response has been sent.

The above code is from the middleware documentation, but it basically means the code for writing the log will be executed before the response is sent.

### Operating System

Linux

### FastAPI Version

0.65.1

### Python Version

3.8.5

**Comment by STeveShary** [comment]
@lsmith77 There isn't a hook or event that I know of in FastAPI or Starlette that would support what you are looking for.  You can probably just create an async task to log your response and just not worry about awaiting it.  Tasks do not have to be awaited to be run: https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

Below is an example:

```python3

import asyncio

@app.middleware("http")
async def write_log(request: Request, call_next):
    loop = asyncio.get_event_loop()
    response = await call_next(request)
    loop.create_task(log_response(response))
    return response


async def log_response(response):
    // ...
    logger.info("Hello World!");
```

**Comment by TheoBabilon** [comment, marked as answer]
Wouldn't [background tasks](https://fastapi.tiangolo.com/tutorial/background-tasks/) work for you?

**Comment by lsmith77** [comment]
@TheoBabilon indeed this sounds exactly like that I should be using
